spec/requests/vms_spec.rb

```diff
@@ -82,6 +82,24 @@ def update_raw_power_state(state, *vms)
                                    [{"vendor" => "openstack"},
                                     {"vendor" => "openstack"}])
     end
+
+    context "with indirect virtual attribute ('hardware.cpu_sockets')" do
```

Comment:
Is there a spec for something like `hardware.host.name` as per your example above?

Reply:
So I did this, but I needed to add support for it so it would allow support for nested attributes.

Left it as a separate commit instead of rebasing, so see the last commit for info.